### PR TITLE
add monitoring

### PR DESCRIPTION
**Pull request by jrwren**

### CODE REVIEW

File: internal/charmstore/server.go [hunk 1]
```diff
@@ -146,6 +147,7 @@ func NewServer(db *mgo.Database, si *SearchIndex, config ServerParams, versions
 	}
 	// Version independent API.
 	handle(srv.mux, "/debug", newServiceDebugHandler(pool, config, srv.mux))
+	handle(srv.mux, "/metrics", prometheus.Handler())
```
Comment (hatched) [inline internal/charmstore/server.go:150]: Well that's convenient!

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1852/
Test FAILed.

**Comment by hatched** [comment]
:+1:

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1853/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1854/
Test PASSed.

File: internal/monitoring/monitoring.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+		Subsystem: "handler",
+		Name:      "request_duration",
+		Help:      "The duration of a web request.",
+	}, []string{"path_pattern"})
+	uploadProcessingDuration = prometheus.NewSummary(prometheus.SummaryOpts{
+		Namespace: "charmstore",
+		Subsystem: "archive",
+		Name:      "processing_duration",
+		Help:      "The processing duration of a charm upload.",
+	})
```
Comment (bac) [inline internal/monitoring/monitoring.go:21]: The units should be specified somewhere unless there is an implicit understanding that they are in seconds. Explicit is still better than implicit.

**Comment by bac** [comment]
:+1: with one change.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1855/
Test PASSed.

**Comment by jrwren** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**